src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -103,8 +111,15 @@ export default class DefaultMessagingSession implements MessagingSession {
     });
   }
 
-  private startConnecting(reconnecting: boolean): void {
-    const signedUrl = this.prepareWebSocketUrl();
+  private async startConnecting(reconnecting: boolean) {
+    // reconnect needs to re-resolve endpoint url, which will also refresh credentials on client if they are expired
+    let endpointUrl = !reconnecting ? this.configuration.endpointUrl : null;
```

Comment:
Since we plan to deprecate the `configuration.endpointURL` I think we should change this logic always try to get the endpoint unless users still pass in the endpoint.
```
if (reconnecting || !this.configuration.endpointURL) {
  // Get endpoint
}
```

Reply:
I believe that is what this is doing ;-P, just doing it in a way so let endpointUrl is defined above.  If it is null in configuration then it will fetch it.